src/cache.c

```diff
@@ -34,23 +35,47 @@ static pthread_mutex_t cf_lock;
  */
 static char *DATA_DIR;
 
+/**
+ * \brief Return user-specified cache directory or the default
+ */
+static char *CacheSystem_get_cache_home()
+{
+    char *xdg_cache_home_default = "/.cache";
+    char *cache_home;
+    if (CONFIG.cache_dir) {
+        return CONFIG.cache_dir;
+    } else {
+        /*
+        *  "XDG_CACHE_HOME/HOME/.cache" > "HOME/.cache"  > "./.cache"
+        */
+        char *xdg_cache_home = getenv("XDG_CACHE_HOME");
+        if (!xdg_cache_home) {
+            char *user_home = getenv("HOME");
+            if (!user_home) {
+                cache_home = path_append(".", xdg_cache_home_default);
```

Comment:
If you plan to do that, you need to update the ``--help``, and update the README.